packages/core/src/workflows/workflow-instance.ts

```diff
@@ -76,9 +100,13 @@ export class WorkflowInstance<TSteps extends Step<any, any, any>[] = any, TTrigg
     this.#mastra = mastra;
 
     this.#runId = runId ?? crypto.randomUUID();
-    this.#onStepTransition = onStepTransition;
+
     this.#onFinish = onFinish;
 
+    this.events = events;
+    this.#makeStepDef = makeStepDef;
```

Comment:
I don't think this is right passing the function in like this.

It starts to feel easier to perhaps just pass the workflow instance in the constructor at this point maybe.

Reply:
yeah, got you
I'll update 👍